src/nvim/tui/tui.c

```diff
@@ -844,7 +844,16 @@ static void fix_terminfo(TUIData *data)
   }
 
   if (STARTS_WITH(term, "xterm") || STARTS_WITH(term, "rxvt")) {
-    unibi_set_if_empty(ut, unibi_cursor_normal, "\x1b[?25h");
+    const char *normal = unibi_get_str(ut, unibi_cursor_normal);
+    if (!normal) {
+      unibi_set_str(ut, unibi_cursor_normal, "\x1b[?25h");
+    } else if (STARTS_WITH(normal, "\x1b[?12l")) {
+      // terminfo typically includes DECRST 12 as part of setting up the normal
+      // cursor, which interferes with the user's control via
+      // NVIM_TUI_ENABLE_CURSOR_SHAPE.  When DECRST 12 is present, skip over
+      // it, but honor the rest of the TI setting.
```

Comment:
> It's just a stop-gap

Understood. To be clear, I was suggesting the entire patch could be this:

    unibi_set_str(ut, unibi_cursor_normal, "\x1b[?25h");

libtickit could be years away, I'm very interested in working out a nice solution for 80% of our users in the very near term.